src/main/java/io/cdap/plugin/servicenow/apiclient/ServiceNowTableAPIClientImpl.java

```diff
@@ -375,45 +238,146 @@ public List<Map<String, Object>> fetchTableRecordsRetryableMode(String tableName
    * @return schema for given ServiceNow table
    */
   @Nullable
-  public Schema fetchServiceNowTableSchema(String tableName, FailureCollector collector) {
+  public Schema fetchTableSchema(String tableName, FailureCollector collector) {
+    Schema schema = null;
+    try {
+      schema = fetchTableSchema(tableName);
+    } catch (OAuthProblemException | OAuthSystemException | RuntimeException e) {
+      LOG.error("Error in connection - {}", e);
+      collector.addFailure("Connection failed. Unable to fetch schema for table " + tableName, null);
```

Comment:
> this hides the actual cause of the error from the user. The exception message should be used somewhere, otherwise there will be no way to tell what went wrong.

Added exception message to the logger. e63c214cf864e664ca24072f21c719c1a777b9c4

> should also add the stacktrace to the failure using `withStacktrace`, it will normally be useful.

**Do we want to add stacktrace to the failure collector?** in order to display the message with the complete stacktrace at the time of validation on the UI?

Reply:
stacktrace will include it in the response so it isn't lost (the UI doesn't show it right now).

Reply:
Included the stack trace in the failure message. f58c78717ea204f30540f24a0352ce21ef7ed47b